status-core/src/main/java/com/indeed/status/core/AbstractDependencyManagerParams.java

```diff
@@ -0,0 +1,52 @@
+package com.indeed.status.core;
+
+import com.indeed.util.core.time.WallClock;
+import org.immutables.value.Value;
+
+import javax.annotation.Nullable;
+import java.util.concurrent.ThreadPoolExecutor;
+
+import static com.indeed.status.core.AbstractDependencyManager.newDefaultThreadPool;
+
+@Value.Immutable
+public abstract class AbstractDependencyManagerParams {
+    @Nullable
+    public abstract String appName();
+
+    @Nullable
+    public abstract String loggerName();
+
+    @Value.Default
+    public ThreadPoolExecutor threadPool() {
+        return newDefaultThreadPool();
+    }
+
+    @Nullable
+    protected abstract WallClock wallClock();
```

Comment:
Can we avoid this? Or at least mark it deprecated? I thought we deprecated `WallClock` in favor of `java.time.Clock`?

Reply:
Seems difficult to deprecate for now. There's no alternative unless we pass in both WallClock and Clock since we have no compatibility layer connecting the two.